### ISSUE TITLE
[java] InvalidLogMessageFormat rule : False positive for lambda parameter

### ISSUE THREAD

**Issue by julius-d** [pmd/pmd#2718]
**Affects PMD Version:** 
PMD version: 6.26.0

**Description:**

In my code the PMD rule [InvalidLogMessageFormat](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#invalidlogmessageformat)
found a bug, that is not a bug.

I was able to reproduce it with this code sample

**Code Sample demonstrating the issue:**

```java
import java.util.Optional;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Foo {

  private static final Logger LOG = LoggerFactory.getLogger(Foo.class);

  public static class Event {

    private Throwable lastThrowable;

    public Throwable getLastThrowable() {
      return lastThrowable;
    }
  }

  public void bar(Event event) {
    Optional.of(event).ifPresent(e -> LOG.warn(
        "Failed on {}",
        1,
        e.getLastThrowable()));

  }
}

```

**Steps to reproduce:**

On my machine:
1.  `mvn clean compile pmd:check`
2.  See `You have 1 PMD violation`
3. `cat target/pmd.xml`
4.  See 
```
<file name="/xyz/Foo.java">
<violation beginline="21" endline="21" begincolumn="39" endcolumn="46" rule="InvalidLogMessageFormat" ruleset="Error Prone" package="xyz" class="Foo" method="bar" externalInfoUrl="https://pmd.github.io/pmd-6.26.0/pmd_rules_java_errorprone.html#invalidlogmessageformat" priority="5">
Too many arguments, expected 1 argument but have 2
</violation>
</file>
```
In this project:
1. Add the snippet to src/test/resources/net/sourceforge/pmd/lang/java/rule/errorprone/xml/InvalidLogMessageFormat.xml
2. Run `mvn test`
3. See 
```
[ERROR] Failures: 
[ERROR] net.sourceforge.pmd.lang.java.rule.errorprone.InvalidLogMessageFormatTest.null
...
[ERROR]   Run 45: InvalidLogMessageFormatTest>RuleTst.runTest:157 "My bug" resulted in wrong number of failures, expected:<0> but was:<1>
```

**Running PMD through:** *Maven*
maven-pmd-plugin:3.13.0
          
Apache Maven 3.6.3 
Java version: 11.0.1, vendor: AdoptOpenJDK, 
platform encoding: UTF-8
OS name: "mac os x"

**Comment by krallus** [comment]
`LOG.warn("Failed on {}", 1, e.getLastThrowable())` does indeed present two arguments when your string only asks for one.  I believe you want one of the methods that ends with a Throwable parameter, but there is no such method as `Logger.warn(String, Object..., Throwable)` (nor would one be possible in that order).  I suggest instead you use `Logger.warn(Message, Throwable)`:
```
LOG.warn(new ParameterizedMessage("Failed on {}", 1), e.getLastThrowable())
```

**Comment by adangel** [comment]
Actually, the method `Logger.warn(String, Object, Object)` is being called. While the last argument here is not throwable, it will still be recognized as such - I only checked logback, but I expect this to be the same for all logger implementations: If the last argument is a throwable, then the arguments are rearranged and the last argument is removed  from the message params and used as what it is: a throwable.
I assume, this is the same for all logger implementations, otherwise it would be very surprising...

So the call `LOG.warn("Failed on {}", 1, e.getLastThrowable()));` is valid and really has only one message parameter and one throwable. The problem here is probably, that PMD is not recognizing the last argument as throwable correctly due to a method call.

**Comment by adangel** [comment]
To help PMD to identify the throwable, you can add an explicit cast, like this: `LOG.warn("Failed on {}", 1, (Throwable) e.getLastThrowable()));`.

**Comment by krallus** [comment]
> While the last argument here is not throwable, it will still be recognized as such

Nice! I would not have guessed that at all from the javadoc.

**Comment by jsotuyod** [comment]
Under PMD 7 this is no longer reproducible.
